digiroad2-oracle/src/main/scala/fi/liikennevirasto/digiroad2/service/linearasset/ManoeuvreService.scala

```diff
@@ -335,40 +342,47 @@ class ManoeuvreService(roadLinkService: RoadLinkService, eventBus: DigiroadEvent
 
-    val (intermediates, adjacents, adjacentConnectPoint) = recursiveGetAdjacent(tsLinkId, connectionPoint)
-    val manoeuvreInit = trafficSignInfo.roadLink +: intermediates
+    try {
+        val (intermediates, adjacents, adjacentConnectPoint) = recursiveGetAdjacent(tsLinkId, connectionPoint)
+        val manoeuvreInit = trafficSignInfo.roadLink +: intermediates
 
-      val tsType = TrafficSignType.applyOTHValue(trafficSignInfo.signType)
+        val tsType = TrafficSignType.applyOTHValue(trafficSignInfo.signType)
 
-      val roadLinks = tsType match {
-        case NoLeftTurn => manoeuvreInit :+ roadLinkService.pickLeftMost(manoeuvreInit.last, adjacents)
+        val roadLinks = tsType match {
+          case NoLeftTurn => manoeuvreInit :+ roadLinkService.pickLeftMost(manoeuvreInit.last, adjacents)
 
-        case NoRightTurn => manoeuvreInit :+ roadLinkService.pickRightMost(manoeuvreInit.last, adjacents)
+          case NoRightTurn => manoeuvreInit :+ roadLinkService.pickRightMost(manoeuvreInit.last, adjacents)
 
-        case NoUTurn =>
-          val firstLeftMost = roadLinkService.pickLeftMost(manoeuvreInit.last, adjacents)
-          val (int, newAdjacents, _)= recursiveGetAdjacent(firstLeftMost.linkId, getOpositePoint(firstLeftMost.geometry, adjacentConnectPoint))
-          (manoeuvreInit :+ firstLeftMost) ++ (int :+ roadLinkService.pickLeftMost(firstLeftMost, newAdjacents))
+          case NoUTurn =>
+            val firstLeftMost = roadLinkService.pickLeftMost(manoeuvreInit.last, adjacents)
+            val (int, newAdjacents, _)= recursiveGetAdjacent(firstLeftMost.linkId, getOpositePoint(firstLeftMost.geometry, adjacentConnectPoint))
+            (manoeuvreInit :+ firstLeftMost) ++ (int :+ roadLinkService.pickLeftMost(firstLeftMost, newAdjacents))
 
-        case _ => Seq.empty[RoadLink]
-      }
+          case _ => Seq.empty[RoadLink]
+        }
 
+      if(!validateManoeuvre(trafficSignInfo.roadLink.linkId, roadLinks.last.linkId, ElementTypes.FirstElement))
+        throw new ManoeuvreCreationException(Set("Manoeuvre creation not valid"))
+      else
+        Seq(createWithoutTransaction("traffic_sign_generated", NewManoeuvre(Set(), Seq.empty[Int], None, roadLinks.map(_.linkId), Some(trafficSignInfo.id), false), roadLinks))
 
-    if(!validateManoeuvre(trafficSignInfo.roadLink.linkId, roadLinks.last.linkId, ElementTypes.FirstElement))
-      throw new ManoeuvreCreationException(Set("Manoeuvre creation not valid"))
-    else
-      Seq(createWithoutTransaction("traffic_sign_generated", NewManoeuvre(Set(), Seq.empty[Int], None, roadLinks.map(_.linkId), Some(trafficSignInfo.id), false), roadLinks))
+    } catch {
+      case mce: ManoeuvreCreationException =>
+          insertInaccurateAsset(trafficSignInfo, fromTrafficSignGenerator, mce)
```

Comment:
What is the reason to get this exception? Something previous seems to allow incorrect operation